src/utils/parse.rs

```diff
@@ -67,6 +71,46 @@ pub(crate) fn parse_matrix_string(string: &str) -> Result<Vec<Vec<String>>, Math
     Ok(matrix)
 }
 
+/// Takes a matrix as input and converts it to a [`String`].
+///
+/// Parameters:
+/// - `matrix`: a matrix e.g. [`MatZ`](crate::integer::MatZ), [`MatQ`](crate::rational::MatQ).
+///
+/// Returns the Matrix in form of a [`String`]. For matrix `[[1, 2, 3],[4, 5, 6]]`
+/// the String looks like this `[[1, 2, 3],[4, 5, 6]]`.
+pub(crate) fn matrix_to_string<T: GetEntry<Z> + GetNumRows + GetNumColumns>(matrix: &T) -> String {
```

Comment:
Are there advantages of choosing a string_builder, I have yet to encounter any. To be consistent with out current implementation, consider getting the entries as a Vector and joining each entry of a row with ", " as a separater and all rows with "],[" it would only remain to add the "[[" at the start and the "]]" at the end.

Reply:
s.o.